src/main/cpp/apple_fsnotifier.cpp

```diff
@@ -40,9 +39,7 @@ static void reportEvent(const char *event, char *path) {
     JNIEnv* env;
     int getEnvStat = jvm->GetEnv((void **)&env, JNI_VERSION_1_6);
     if (getEnvStat == JNI_EDETACHED) {
-        printf("~~~~ GetEnv: not attached");
         if (jvm->AttachCurrentThread((void **) &env, NULL) != 0) {
-            printf("~~~~ Failed to attach");
```

Comment:
Let's add `// TODO`-style comments here to make sure we don't forget error handling.

Reply:
I've added error handling using `FunctionResult` object.